must-gather/gather-kepler-operator-info

```diff
@@ -0,0 +1,92 @@
+source "${SCRIPT_DIR}/common"
+
+BASE_COLLECTION_PATH=$1
+
+KEPLER_OPERATOR_NS="openshift-operators"
+
+get_catalogsource() {
+  log "collecting catalogsource info for kepler-operator"
+  run oc -n "$KEPLER_OPERATOR_NS" get catalogsource kepler-operator-catalog -oyaml "$KEPLER_OPERATOR_INFO_DIR/kepler-operator-catalogsource.yaml"
+}
+
+get_subscription() {
+  log "collecting subscription info for kepler-operator"
+  run oc -n "$KEPLER_OPERATOR_NS" get subscription -l operators.coreos.com/kepler-operator.openshift-operators= -oyaml "$KEPLER_OPERATOR_INFO_DIR/kepler-operator-subscription.yaml"
+}
+
+get_install_plan() {
+  log "collecting installplan info for kepler-operator"
+  run oc -n "$KEPLER_OPERATOR_NS" get installplan -l operators.coreos.com/kepler-operator.openshift-operators= -oyaml "$KEPLER_OPERATOR_INFO_DIR/kepler-operator-installplan.yaml"
+}
+
+get_csv() {
+  log "collecting CSV for kepler-operator"
+  run oc -n "$KEPLER_OPERATOR_NS" get csv -l operators.coreos.com/kepler-operator.openshift-operators= -oyaml "$KEPLER_OPERATOR_INFO_DIR/kepler-operator-csv.yaml"
+}
+
+get_kepler_operator_deployment_info() {
+  KEPLER_OPERATOR_DEPLOY="kepler-operator-controller-manager"
+  log "collecting deployment info for kepler-operator"
+  run oc -n "$KEPLER_OPERATOR_NS" get deployment "$KEPLER_OPERATOR_DEPLOY" -oyaml "$KEPLER_OPERATOR_INFO_DIR/kepler-operator-deployment.yaml"
+}
+
+get_kepler_operator_pod_info() {
+  log "collecting pod info for kepler-operator" 
+  run oc -n "$KEPLER_OPERATOR_NS" get pod -l control-plane=controller-manager -oyaml "$KEPLER_OPERATOR_INFO_DIR/kepler-operator.yaml"
+}
+
+get_summary() {
+  run oc -n "$KEPLER_OPERATOR_NS" get catalogsource kepler-operator-catalog -owide "$KEPLER_OPERATOR_INFO_DIR/summary.txt"
+  echo -e "\n" >> "$KEPLER_OPERATOR_INFO_DIR/summary.txt"
+
+  run oc -n "$KEPLER_OPERATOR_NS" get subscription -owide "$KEPLER_OPERATOR_INFO_DIR/summary.txt"
+  echo -e "\n" >> "$KEPLER_OPERATOR_INFO_DIR/summary.txt"
+
+  run oc -n "$KEPLER_OPERATOR_NS" get installplan -owide "$KEPLER_OPERATOR_INFO_DIR/summary.txt"
+  echo -e "\n" >> "$KEPLER_OPERATOR_INFO_DIR/summary.txt"
+
+  run oc -n "$KEPLER_OPERATOR_NS" get csv -owide  "$KEPLER_OPERATOR_INFO_DIR/summary.txt"
+  echo -e "\n" >> "$KEPLER_OPERATOR_INFO_DIR/summary.txt"
+
+  KEPLER_OPERATOR_DEPLOY="kepler-operator-controller-manager"
+  run oc -n "$KEPLER_OPERATOR_NS" get deployment "$KEPLER_OPERATOR_DEPLOY" -owide "$KEPLER_OPERATOR_INFO_DIR/summary.txt"
+  echo -e "\n" >> "$KEPLER_OPERATOR_INFO_DIR/summary.txt"
+
+  run oc -n "$KEPLER_OPERATOR_NS" get pod -l control-plane=controller-manager -owide "$KEPLER_OPERATOR_INFO_DIR/summary.txt"
+}
+
+main() {
+
+  KEPLER_OPERATOR_INFO_DIR="$BASE_COLLECTION_PATH/kepler-operator-info"
+  mkdir -p "$KEPLER_OPERATOR_INFO_DIR"
+
+  get_subscription
+  get_catalogsource
+  get_install_plan
+  get_csv
+  get_kepler_operator_deployment_info
+  get_kepler_operator_pod_info
+  get_summary
```

Comment:
should we add || true to every step ?

Reply:
`set -e` should have stopped the script.

Reply:
good that i didn't add it then :)

Reply:
I will verify this but we should always ensure that these steps break (exit with non-zero value)  due to the nature of bash. E.g. say there is a typo in any of the command above, we should be able to find that outduring development we would comment out `# || true` in the optional steps. `get_catalog_source` should be optional since that is will only be present if you `run bundle`

Reply:
if we have to deal with so much complexity, i would rather have must-gather written as a go program. `oc adm must-gather` does not care if it is a script or a binary.